### PR TITLE
refactor: moved passage asset encoding out of SharedStimulusPackageIm…

### PR DESCRIPTION
**Pull request by pnal**
Related PRs:
- https://github.com/oat-sa/extension-tao-nfer-enot/pull/3
- https://github.com/oat-sa/extension-tao-mediamanager/pull/350

### CODE REVIEW

File: model/qti/asset/handler/SharedStimulusAssetHandler.php [hunk 1]
```diff
@@ -207,4 +211,9 @@ public function finalize()
     {
         // Nothing to do
     }
+
+    private function getSharedStimulusMediaEncoderService (): SharedStimulusMediaEncoder
+    {
+        return $this->getServiceLocator()->get(SharedStimulusMediaEncoder::class);
```
Comment (pnal) [inline model/qti/asset/handler/SharedStimulusAssetHandler.php:217]: todo: use SERVICE_KEY

**Review by andreluizmachado — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by gabrielfs7 — APPROVED** [review]

File: model/qti/asset/handler/SharedStimulusAssetHandler.php [hunk 2]
```diff
@@ -207,4 +211,9 @@ public function finalize()
     {
         // Nothing to do
     }
+
+    private function getSharedStimulusMediaEncoderService (): SharedStimulusMediaEncoder
+    {
+        return $this->getServiceLocator()->get(SharedStimulusMediaEncoder::SERVICE_ID);
```
Comment (shpran) [inline model/qti/asset/handler/SharedStimulusAssetHandler.php:217]: Please, use PSR container:
```php
return $this->getServiceLocator()->getContainer()->get(SharedStimulusMediaEncoder::SERVICE_ID);
```

File: model/qti/asset/handler/SharedStimulusAssetHandler.php [hunk 3]
```diff
@@ -207,4 +211,9 @@ public function finalize()
     {
         // Nothing to do
     }
+
+    private function getSharedStimulusMediaEncoderService (): SharedStimulusMediaEncoder
```
Comment (shpran) [inline model/qti/asset/handler/SharedStimulusAssetHandler.php:215]:
```suggestion
    private function getSharedStimulusMediaEncoderService(): SharedStimulusMediaEncoder
```

**Review by hectoras — APPROVED** [review]

**Pull request merged**